src/components/authentication/connect-wallet.tsx

```diff
@@ -0,0 +1,76 @@
+import { type FC, useEffect, useRef } from "react";
+import { useRouter } from "next/router.js";
+
+import { useBBWallet } from "~/service/bb-wallet.js";
+import { type LoggedUserData } from "~/types/index.js";
+import { pages } from "~/router.js";
+
+import { LoginLoading } from "./login-loading.jsx";
+import { type LoginStage } from "./types.js";
+
+interface ConnectWalletProps {
+  loggedUser?: LoggedUserData;
+  setAddress: (address: string) => void;
+  setSignature: (signature: string) => void;
+  setLoginStage: (stage: LoginStage) => void;
+}
+
+export const ConnectWallet: FC<ConnectWalletProps> = ({
+  loggedUser,
+  setAddress,
+  setSignature,
+  setLoginStage,
+}) => {
+  const router = useRouter();
+
+  const { connect, isConnecting, isSigning, getSignature, address } =
+    useBBWallet();
+
+  const isAuthenticating = useRef(false);
+
+  useEffect(() => {
+    const authenticate = async () => {
+      if (isConnecting || isSigning || isAuthenticating.current) return;
+
+      isAuthenticating.current = true;
+
+      try {
+        if (!address) {
+          await connect();
+        } else {
+          const signature = await getSignature();
+
+          setAddress(address);
```

Comment:
Don't you want to set the address only in case the signature when well?

Reply:
If the signature fails we get redirected out of the page anyway so it does not matter

Reply:
The flow gets triggered automatically on the Login Page. The second anon is pretty well explained in the rectangle: in one scenario it has the username and not the address, in the second scenario it lacks both. Theoretically both are possible and need to be handled so I'm specifying these details for clarity.